src/main.rs

```diff
@@ -3,26 +3,58 @@ extern crate lazy_static;
 
 mod kaenguru;
 mod xd;
+mod config;
+mod replies;
 
-use std::borrow::Borrow;
-use std::env;
-use std::panic;
-use std::process::exit;
-use std::thread::sleep;
-use std::time::Duration;
-use tokio::runtime::Runtime;
-
+use std::{
+    borrow::Borrow,
+    env,
+    process::exit,
+    thread::sleep,
+    time::Duration,
+    sync::{Arc},
+};
 use serenity::prelude::*;
-use tokio::time::Instant;
+use tokio::{
+    runtime::Runtime,
+    time::Instant,
+};
+
+use crate::config::CONFIG;
 
 /// Function to start a new instance of the autokommentator bot
 async fn start_xd() {
-    panic::set_hook(Box::new(|_| {
-        eprintln!("Fatal: Not discord token XD found!\nFatal: Please set the DISCORD_TOKEN_XD environment variable to your discord token!\nFatal: More information can be found here: https://mafelp.github.io/MCDC/installation#get-a-discord-bot-token");
-        exit(2);
-    }));
-    let xd_token = env::var("DISCORD_TOKEN_XD").expect("xd_token");
-    let _ = panic::take_hook();
+    // Get the token from the configuration
+    let xd_token = match env::var("DISCORD_TOKEN_XD") {
+        Ok(s) => s,
+        Err(_) => {
+            let config_arc = Arc::clone(&CONFIG);
+            let mut config_lock = config_arc.lock();
+            while config_lock.is_err() {
```

Comment:
A error in locking means the lock has been poisoned and can't be unlocked again.
This should panic instead of retrying!